tests/unit/peapods/pods/test_compound_pods.py

```diff
@@ -162,3 +162,54 @@ def test_pod_naming_with_parallel(runtime):
         assert bp.replica_list[2].peas[1].runtime.name == 'pod/2/tail/ZEDRuntime'
         assert bp.replica_list[2].peas[2].runtime.name == 'pod/2/0/ZEDRuntime'
         assert bp.replica_list[2].peas[3].runtime.name == 'pod/2/1/ZEDRuntime'
+
+
+@pytest.mark.parametrize(
+    'num_hosts, used_hosts',
+    (
+        (
+            6,
+            (
+                ['0.0.0.1', '0.0.0.2'],
+                ['0.0.0.3', '0.0.0.4'],
+                ['0.0.0.5', '0.0.0.6'],
+            ),
+        ),
+        (
+            8,
+            (
+                ['0.0.0.1', '0.0.0.2'],
+                ['0.0.0.3', '0.0.0.4'],
+                ['0.0.0.5', '0.0.0.6'],
+            ),
+        ),
+        (
+            3,
+            (
+                ['0.0.0.1', '0.0.0.2'],
+                ['0.0.0.3', '0.0.0.1'],
+                ['0.0.0.2', '0.0.0.3'],
+            ),
+        ),
+    ),
+)
+def test_host_list_matching(num_hosts, used_hosts):
+    args = set_pod_parser().parse_args(
+        [
+            '--name',
+            'pod',
+            '--parallel',
+            '2',
+            '--replicas',
+            '3',
+            '--peas-hosts',
+            *[f'0.0.0.{i+1}' for i in range(num_hosts)],
+            '--runtime-backend',
+            'process',
+        ]
+    )
```

Comment:
Can you try to convert one of the functional integration tests (CRUD etc.) to use this approach, to see that the ops work properly in this new setup?